parser/parser/cc/grammars/typedruby.ypp

```diff
@@ -2461,16 +2463,20 @@ array_premature_end: eof
               do: term
                 | kDO_COND
 
-         if_tail: opt_else
+         if_tail: opt_else kend_or_eof
```

Comment:
kend_or_eof` is the special production rule that powers the indentationAware parser recovery mode.

If we've pushed it into `if_tail`, then all the `if_tail` rules that were followed only by `kEND` instead of `kend_or_eof` are now better at parse error recovery. It would be good to spend some time trying to search for such a test case.